docs/backtracking algorithms/N-QueensProblem.md

```diff
@@ -0,0 +1,146 @@
+# N-Queens Problem
```

Comment:
Please follow our format and syntax

Reply:
I have updated the code based on your suggestions. please check it out